src/modules/facility-loan/accrual-schedule.builder.test.ts

```diff
@@ -0,0 +1,164 @@
+import { ENUMS } from '@ukef/constants';
+import { CURRENCIES } from '@ukef/constants/currencies.constant';
+import { LOAN_RATE_INDEX } from '@ukef/constants/loan-rate-index.constant';
+import { CurrentDateProvider } from '@ukef/modules/date/current-date.provider';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+import { TEST_DATES } from '@ukef-test/support/constants/test-date.constant';
+import { CreateFacilityLoanGenerator } from '@ukef-test/support/generator/create-facility-loan-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+
+import { AccrualScheduleBuilder } from './accrual-schedule.builder';
+
+describe('AccrualScheduleBuilder', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const facilityIdentifier = valueGenerator.facilityId();
+  const bundleIdentifier = valueGenerator.acbsBundleId();
+  const dateStringTransformations = new DateStringTransformations();
+  const currentDateProvider = new CurrentDateProvider();
+
+  let accrualScheduleBuilder: AccrualScheduleBuilder;
+
+  beforeEach(() => {
+    accrualScheduleBuilder = new AccrualScheduleBuilder(dateStringTransformations, currentDateProvider);
+  });
+
+  describe('getAccrualSchedules', () => {
+    const {
+      requestBodyToCreateFacilityLoanGbp,
+      requestBodyToCreateFacilityLoanNonGbp,
+      bondAndGefAccrualSchedulesGbp,
+      ewcsAccrualSchedulesUsd,
+      ewcsAccrualSchedulesGbp,
+    } = new CreateFacilityLoanGenerator(valueGenerator, dateStringTransformations).generate({
+      numberToGenerate: 1,
+      facilityIdentifier,
+      bundleIdentifier,
+    });
+    const [newLoanGbp] = requestBodyToCreateFacilityLoanGbp;
+    const [newLoanNonGbp] = requestBodyToCreateFacilityLoanNonGbp;
+
+    it(`generates bond repayment schedules when product type is 'BOND'`, () => {
+      const newLoanWithProductTypeGroupBondCurrencyGbp = {
+        ...newLoanGbp,
+        productTypeGroup: ENUMS.PRODUCT_TYPE_GROUPS.BOND,
+      };
+      const accrualSchedules = accrualScheduleBuilder.getAccrualSchedules(newLoanWithProductTypeGroupBondCurrencyGbp);
+
+      expect(accrualSchedules).toEqual(bondAndGefAccrualSchedulesGbp);
+    });
+
+    it(`generates gef repayment schedules when product type is 'GEF'`, () => {
+      const newLoanWithProductTypeGroupBondCurrencyGbp = {
+        ...newLoanGbp,
+        productTypeGroup: ENUMS.PRODUCT_TYPE_GROUPS.GEF,
+      };
+      const accrualSchedules = accrualScheduleBuilder.getAccrualSchedules(newLoanWithProductTypeGroupBondCurrencyGbp);
+
+      expect(accrualSchedules).toEqual(bondAndGefAccrualSchedulesGbp);
+    });
```

Comment:
You could probably do the following with the highlighted and the test on line 90
```
    it.each([
      { productTypeGroup: ENUMS.PRODUCT_TYPE_GROUPS.BOND, expectedResult: bondAndGefAccrualSchedulesGbp },
      { productTypeGroup: ENUMS.PRODUCT_TYPE_GROUPS.GEF, expectedResult: bondAndGefAccrualSchedulesGbp },
      { productTypeGroup: ENUMS.PRODUCT_TYPE_GROUPS.EWCS, expectedResult: ewcsAccrualSchedulesGbp },
    ])('generates $productTypeGroup repayment schedule', ({ productTypeGroup, expectedResult }) => {
      const newLoanWithProductTypeGroup = {
        ...newLoanGbp,
        productTypeGroup: productTypeGroup,
      };
      const accrualSchedules = accrualScheduleBuilder.getAccrualSchedules(newLoanWithProductTypeGroup);

      expect(accrualSchedules).toEqual(expectedResult);
    });
```

Reply:
It does tidy up these tests but I think I'll keep EWCS separate because there are a few different outcomes dependent on currency which I've grouped together instead

Reply:
This does tidy these tests up but I might keep the EWCS test separate as there is some logic which is `currency` dependent so have grouped those tests together instead